azure/converters/extendedlocation.go

```diff
@@ -17,12 +17,27 @@ limitations under the License.
 package converters
 
 import (
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
 	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 )
 
+// ExtendedLocationToNetworkSDKv2 converts an infrav1.ExtendedLocationSpec to an armnetwork.ExtendedLocation.
+func ExtendedLocationToNetworkSDKv2(src *infrav1.ExtendedLocationSpec) *armnetwork.ExtendedLocation {
+	if src == nil {
+		return nil
+	}
+	if src.Type == "Edge" {
+		src.Type = "EdgeZone"
+	}
```

Comment:
Out of curiosity, when might we expect to reach this case? It looks like the CAPZ API has only ever allowed `EdgeZone` as a valid value here.

Reply:
I think I was led astray by the existing (SDK v1) unit test that uses "Edge" as a test value. But it doesn't appear we'll see this in practice, so I removed this logic.